lib/src/over_react_redux/over_react_flux.dart

```diff
@@ -392,197 +570,33 @@ UiFactory<TProps> Function(UiFactory<TProps>)
-/// have Redux components, this class should not be used (implementing
-/// [FluxToReduxAdapterStore] instead) and the Redux boilerplate should be added.
-/// If needed, however, a `connect`ed component can receive updates from
-/// this implementation (just not trigger them).
-///
-/// __Example:__
-/// ```dart
-/// import 'package:w_flux/w_flux.dart' as flux;
-///
-/// class ExampleFluxStore extends flux.Store {
-///   FluxActions _actions
-///
-///   String _valueFromState = 'Default Value';
-///   String get valueFromState => _valueFromState;
-///
-///   ExampleFluxStore(this._actions) {
-///     triggerOnActionV2(_actions.fluxAction, _mutateValueFromState);
-///   }
-///
-///   void _mutateValueFromState(String newValue) {
-///     _valueFromState = newValue;
-///   }
-/// }
-///
-/// class FluxActions {
-///   final flux.Action<String> changeBackgroundColor = flux.Action();
-/// }
-///
-/// FluxActions actions = FluxActions();
-/// ExampleFluxStore store = ExampleFluxStore(actions);
-///
-/// ConnectFluxAdapterStore adapterStore = ConnectFluxAdapterStore<ExampleFluxStore, Null>(store, actions);
-/// ```
-///
-/// Related: [connectFlux], [FluxToReduxAdapterStore]
-class ConnectFluxAdapterStore<S extends flux.Store> extends redux.Store<S> {
-  /// A reference to an instantiated Flux store object that backs the adapter store.
-  ///
-  /// This store instance is the actual container of all store data, with the
-  /// [ConnectFluxAdapterStore] acting as an interface to connect to the store.
-  /// Unlike a [FluxToReduxAdapterStore], updates to the store should come from Flux.
-  final S store;
-
-  StreamSubscription _storeListener;
-
-  ConnectFluxAdapterStore(this.store, dynamic actions,
-      {List<redux.Middleware<S>> middleware})
-      : super((_, __) => store,
-            middleware: middleware ?? const [],
-            initialState: store,
-            distinct: false) {
-    assert(store is! InfluxStoreMixin,
-        'Use FluxToReduxAdapterStore when your store implements InfluxStoreMixin');
-
-    _storeListener = store.listen((_) {
-      dispatch(_FluxStoreUpdatedAction());
-    });
-
-    actionsForStore[store] = actions;
-  }
-
-  @override
-  Future teardown() async {
-    await _storeListener.cancel();
-    await super.teardown();
-  }
-}
-
 extension InfluxStoreExtension<S extends InfluxStoreMixin> on S {
   /// Returns a [FluxToReduxAdapterStore] instance from the Flux store instance.
   ///
   /// This is meant to be a more succinct way to instantiate the adapter store.
   FluxToReduxAdapterStore asReduxStore(dynamic actions,
       {List<redux.Middleware> middleware}) {
-    return FluxToReduxAdapterStore(this, actions, middleware: middleware);
+    if (_storeAdapterFor[this] != null) return _storeAdapterFor[this];
+
+    _storeAdapterFor[this] = FluxToReduxAdapterStore(this, actions, middleware: middleware);
+    return _storeAdapterFor[this];
```

Comment:
#nit this method could be simplified into a single statement using null-awares:
```dart
FluxToReduxAdapterStore asReduxStore(dynamic actions,
        {List<redux.Middleware> middleware}) =>
    _storeAdapterFor[this] ??=
        FluxToReduxAdapterStore(this, actions, middleware: middleware);
```
(same below)